### ISSUE TITLE
Regarding Do intervention on values other than the training set

### ISSUE THREAD

**Issue by AizazSharif** [quantumblacklabs/causalnex#165]
Hi,

I have been using Causalnex library over the past month in my PhD and its amazing even with large number of features. 
I am currently exploring Inference engine for do-intervention step and I realized that you can only intervene on values that already exists in the CPD of Bayesian model. 

Is there any method to perform counterfactual reasoning on unknown data (closer to the training data values) which we want to use for do-intervention? 
Any help or suggestion would be appreciated.

**Comment by GabrielAzevedoFerreiraQB** [comment]
Hi @AizazSharif , and thanks for the question!

Indeed, since CausalNex works with a discretised version of the data (see the discretisation methods [here](https://github.com/quantumblacklabs/causalnex/tree/develop/causalnex/discretiser) ), The Bayesian Model can't evaluate a value that is not present in the training data.

There are alternatives to this:

- You can specify your own CPD/ edit the CPD to contain the extra values you want to use for querying. This would be a somehow manual process, though and you would have to judge what are the best probabilities for the unseen value - since we can't infer it from data.
- It might be worth while looking at our [Latent Values ](https://github.com/quantumblacklabs/causalnex/blob/develop/causalnex/estimator/em.py)module. It essentially allows you to add entire variables that the data doesn't account for. it is one level different from your use case, but it may help. The advantage here is that the model can learn the probabilities even without data.

I hope it helps!

**Comment by oentaryorj** [comment]
Closing this issue for now. Feel free to re-open should there be further question
